bindings/java/src/integration/com/apple/foundationdb/RequiresDatabase.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * RequiresDatabase.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.AssumptionViolatedException;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Rule to make it easy to write integration tests that only work when a running
+ * database is detected and connectable using the default cluster file. Use this
+ * as a @ClassRule on any integration test that requires a running database.
+ * 
+ * This will attempt to connect to an FDB instance and perform a basic operation. If it can
+ * do so quickly, then it will go ahead and run the underlying test statement. If it cannot 
+ * perform a basic operation against the running DB, then it will throw an AssumptionViolatedException
+ * and skip running the test (JUnit treats AssumptionViolatedException as a skipped test when recording data).
+ * 
+ * There is a second safety valve--you can also set the env variable `run.integration.tests` to false. If it's set,
+ * then all tests will just be skipped outright, without trying to connect.
```

Comment:
I would prefer only having the environment variable method. I think it is reasonable to assume the process running tests knows whether or not there is a cluster available.

Reply:
the env. variable should be thought of as the main mechanism. I converted skipping to failing the test, because when empirically running the tests without a server running felt like the tests were hanging (a fail-fast behavior had cleaner ergonomics).

Reply:
it doesn't seem to me like skipping the test is fail-fast, just because it means the suite could pass because the local db was hosed. Won't the tests themselves or the transactions that they set up, have time limits? That seems closer to how the API gets used.